src/avt_vimba_camera.cpp

```diff
@@ -111,33 +99,33 @@ void AvtVimbaCamera::start(const std::string& ip_str, const std::string& guid_st
       vimba_camera_ptr_->GetSerialNumber(cam_guid_str);
       if (!vimba_camera_ptr_)
       {
-        ROS_WARN("Camera pointer is empty. Returning...");
+        RCLCPP_WARN(nh_->get_logger(), "Camera pointer is empty. Returning...");
         return;
       }
       assert(cam_guid_str == guid_str);
       updater_.setHardwareID(guid_str);
       guid_ = guid_str;
       diagnostic_msg_ = "GUID " + cam_guid_str + " matches for camera with IP: " + ip_str;
-      ROS_INFO_STREAM("GUID " << cam_guid_str << " matches for camera with IP: " << ip_str);
+      RCLCPP_INFO_STREAM(nh_->get_logger(), "GUID " << cam_guid_str << " matches for camera with IP: " << ip_str);
     }
   }
   else if (!guid_str.empty())
   {
     // Only guid available
     diagnostic_msg_ = "Trying to open camera by ID: " + guid_str;
-    ROS_INFO_STREAM("Trying to open camera by ID: " << guid_str);
-    vimba_camera_ptr_ = openCamera(guid_str, print_all_features);
+    RCLCPP_INFO_STREAM(nh_->get_logger(), "Trying to open camera by ID: " << guid_str);
+    vimba_camera_ptr_ = openCamera(guid_str);
     updater_.setHardwareID(guid_str);
     guid_ = guid_str;
   }
   else
   {
     // No identifying info (GUID and IP) are available
     diagnostic_msg_ = "Can't connect to the camera: at least GUID or IP need to be set.";
-    ROS_ERROR("Can't connect to the camera: at least GUID or IP need to be set.");
+    RCLCPP_ERROR(nh_->get_logger(), "Can't connect to the camera: at least GUID or IP need to be set.");
     camera_state_ = ERROR;
   }
-  updater_.update();
+  updater_.force_update();
 
   getFeatureValue("GevTimestampTickFrequency", vimba_timestamp_tick_freq_);
 
```

Comment:
In line 145 we compare trigger_source_int to enum. I think that it would be more simple to just std::find(std::begin(TriggerMode), std::end(TriggerMode), trigger_source). We would get rid of this enum, and also USB cameras will be working out of the box.

Reply:
Agreed! I'll fix this.

Reply:
I'm actually going to remove the if statement completely, so it just always creates a FrameObserver. We can leave it up to the user to make sure they have their TriggerSource set up correctly.

Reply:
Great, that would simplify the code.

Reply:
Updated.